### ISSUE TITLE
[Monitoring v2] prometheus rules reset on upgrade

### ISSUE THREAD

**Issue by slickwarren** [rancher/rancher#32225]
**What kind of request is this (question/bug/enhancement/feature request):**
Bug

**Steps to reproduce (least amount of steps as possible):**
* deploy monitoring
* add custom prometheus rule
* upgrade monitoring 

**Result:**
* new custom rule is gone, but built in rules remain

**Other details that may be helpful:**

**Environment information**
- Rancher version (`rancher/rancher`/`rancher/server` image tag or shown bottom left in the UI): v2.5-head (31f00c1) monitoring v14.5.100-rc7
- Installation option (single install/HA): single

**Cluster information**
- Cluster type (Hosted/Infrastructure Provider/Custom/Imported): linode
- Machine type (cloud/VM/metal) and specifications (CPU/memory): 8vCPU
- Kubernetes version (use `kubectl version`): 1.20.5

- Docker version (use `docker version`): 20.10.5

**Comment by slickwarren** [comment]
upgrading a 14.5.100 -> 14.5.100 does keep prom rules through the upgrade, however:

reopening for the following:
* deploy a downstream cluster
* deploy 9.4.20x monitoring
* create some prometheus rules
* upgrade monitoring to 14.5.100

prometheus rules are removed

**Comment by aiyengar2** [comment]
@slickwarren we should also check ServiceMonitors and PodMonitors before resolving this issue.

**Comment by aiyengar2** [comment]
Unable to reproduce. I tried this a couple of times using the following steps on a Rancher 2.5.7 cluster that updated the Charts Repository to point to `dev-v2.5`:
- Deployed a downstream cluster (RKE1, 2 node cluster with one node with All and one node with Worker)
- Installed Monitoring 9.4.200
- Added the below monitoring.coreos.com resources + a namespace for them
- Upgraded 1 major version at a time and checked resources throughout the upgrade steps

Result: Resources always exist. Never get reset on an upgrade.

Perhaps this was resolved / tested on a version of Monitoring 14.5.100 before https://github.com/rancher/charts/pull/1131 was merged? Before this, this problem would have definitely have been an issue.

Alternatively, it could be some combination of 9.4.2xx -> 14.5.100 or something in Rancher master-head / 2.5-head that could have caused the issue, but I'm not sure about why that would be the case.

---

### Resources

```yaml
apiVersion: v1
kind: Namespace
metadata:
  name: example-namespace
---
apiVersion: monitoring.coreos.com/v1
kind: ServiceMonitor
metadata:
  name: example-app-sm
  namespace: example-namespace
  labels:
    app: rancher-monitoring
    release: rancher-monitoring
spec:
  selector:
    matchLabels:
      app: example-app-sm
  endpoints:
  - port: web
---
apiVersion: monitoring.coreos.com/v1
kind: PodMonitor
metadata:
  name: example-app-pm
  namespace: example-namespace
  labels:
    team: frontend
spec:
  selector:
    matchLabels:
      app: example-app-pm
  podMetricsEndpoints:
  - port: web
---
apiVersion: monitoring.coreos.com/v1
kind: PrometheusRule
metadata:
  name: example-app-pr
  namespace: example-namespace
  labels:
    app: rancher-monitoring
    release: rancher-monitoring
spec:
  groups:
  - name: example.rules
    rules:
    - alert: ExampleAlert
      annotations:
        message: Hello World!
      expr: vector(1)
      labels:
        severity: none
        job: example
```

**Comment by slickwarren** [comment]
interesting, I also could not reproduce this on v2.5.8-rc10. Closing this for now
